src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -614,4 +614,25 @@ private String upper(String property, String culture) {
                         "The following method is not supported in Schema: %s",
                         functionName));
     }
+
+    public static Schema.Type getCastFunctionReturnType(String castType) {
+        switch (castType) {
```

Comment:
Should we do `StringUtils.toUpperCase`? Please check if the data type name is case sensitive or not in SQL.

Reply:
Good catch- just tested and need to add, will do that now.